### PR TITLE
feat(policy): add HH-GH-010 contractor repository access control

### PR DESCRIPTION
**Pull request by sdh100shaun**
Introduces a new declarative policy control (HH-GH-010) that identifies
org members whose email matches a configured contractor domain and flags
any pull-request activity in repositories outside the permitted list.

Key changes:
- policies/catalog.yaml: new HH-GH-010 control (family: Access Control)
  with contractor_domains and allowed_repos parameters; maps to NIST
  AC-2/AC-3/AC-6 and CIS Controls 6.1/6.2
- policies/default.yaml + strict.yaml: include HH-GH-010; strict profile
  escalates severity to critical
- src/policy/types.ts: add contractor-repo-access detector type and
  optional orgMembers to EvaluationContext
- src/policy/engine.ts: accept optional orgMembers in evaluate() and
  forward it into the context
- src/types/index.ts: add contractor-repo-access to SecurityIssue type
- src/evaluators/contractor-repo-access-evaluator.ts: evaluator
  implementation with domain matching, allowed-repo enforcement, and
  per-(contractor, repo) deduplication
- src/services/copilot-service.ts: fallback explanation for new type
- 21 unit tests covering happy path, edge cases, deduplication, and
  multi-domain matching

https://claude.ai/code/session_01ThRxVw7BFmLoHmZGVLShwj

### CODE REVIEW

**Comment by sdh100shaun** [comment]
@claude fix the issues in https://github.com/sdh100shaun/HubHelper/pull/24#discussion_r3151442368 and https://github.com/sdh100shaun/HubHelper/pull/24#discussion_r3151442354

File: src/policy/engine.ts [hunk 1]
```diff
@@ -67,6 +74,7 @@ export class PolicyEngine {
       workflowRuns,
       scope: this.policy.scope,
       classifierResults: new Map(),
+      orgMembers,
     };
```
[the remark this answers is not in the document]
Comment (sdh100shaun) [reply src/policy/engine.ts:78]: @copilot apply changes based on [this feedback](https://github.com/sdh100shaun/HubHelper/pull/24#discussion_r3151442368)

File: src/types/index.ts [hunk 2]
```diff
@@ -76,7 +76,8 @@ export interface SecurityIssue {
     | 'disabled-workflow'
     | 'action_failure'
     | 'repeated_action_failure'
-    | 'security-pr-volume';
+    | 'security-pr-volume'
+    | 'contractor-repo-access';
```
[the remark this answers is not in the document]
Comment (sdh100shaun) [reply src/types/index.ts:80]: @copilot apply changes based on [this feedback](https://github.com/sdh100shaun/HubHelper/pull/24#discussion_r3151442354)
